lib/chef/provider/user/windows.rb

```diff
@@ -39,12 +39,12 @@ def load_current_resource
             logger.warn("The 'gid' (or 'group') property is not implemented on the Windows platform. Please use the `members` property of the  'group' resource to assign a user to a group.")
           end
 
-          @current_resource = Chef::Resource::User.new(new_resource.name)
+          @current_resource = Chef::Resource::User::WindowsUser.new(new_resource.name)
```

Comment:
Note: I need to replace this I have added full_name for windows_user resource as it needs only be added for Winodws due to which its not available in base/parent class. So I need to initialise @current_resource with WindowsUser class.

Reply:
you've actually done the right thing here.  current_resource should have been a WindowsUser resource.